src/Pages/Overview/Overview.js

```diff
@@ -34,15 +43,56 @@ export default class OverviewBoard extends Component {
     if (!apiResponse.error) this.setState({ users: apiResponse.responseData })
   }
 
-  updateQuery (event) {
-    let { value } = event.target
+  updateQuery (value) {
+    // taking care of empty values
+    value = typeof value === 'undefined' ? '' : value
     value = value.trim().toLowerCase()
-    this.setState({ query: value })
+    let queryResult
 
-    if (!value) return
-    const queryResult = this.state.users.filter(data =>
-      data.firstName.toLowerCase().includes(value)
-    )
+    const userExists = user => {
+      return (
+        user.firstName.toLowerCase().includes(value) ||
+        user.lastName.toLowerCase().includes(value) ||
+        user.email.toLowerCase().includes(value)
+      )
+    }
+
+    const { currentQueryType } = this.state
+    let filtered, search
+    switch (currentQueryType) {
```

Comment:
to reduce the complexity of this function, let's move the `switch` statement to it's own function e.g.

```js
fiterUserByAccessLevel (accessLevel) {
  switch(accessLevel) {
    case 'Officer':
      return this.state.users.filter(
          data => data.accessLevel === membershipState.OFFICER
        )
    case 'Admin':
      // do admin here
     // other cases here
    default:
      return this.state.users
  }
}
```

then calling on it would look like
```js
let filteredUsersByLevel = fiterUserByAccessLevel(currentQueryType)
let searchResult = filteredUsersByLevel.filter(data => userExists(data))
return searchResult.length ? searchResult : filteredUsersByLevel
```

Reply:
done